File: app/src/main/java/com/learning/ruslan/databases/SupportCursorWrapper.java
package com.learning.ruslan.databases;

import android.database.Cursor;
import android.database.CursorWrapper;

import com.learning.ruslan.Settings;
import com.learning.ruslan.databases.RusDbSchema.SupportTable.Cols;

public class SupportCursorWrapper extends CursorWrapper {

    public SupportCursorWrapper(Cursor cursor) {
        super(cursor);
    }

    public Settings getSettings() {
        Settings settings = new Settings();

        settings.set(
                getInt(getColumnIndex(Cols.CHECKED)) == 1,
                getInt(getColumnIndex(Cols.PAUSE)),
                getString(getColumnIndex(Cols.THEME)),
                getInt(getColumnIndex(Cols.QUESTIONS)),
                getString(getColumnIndex(Cols.LANGUAGE))
        );

        settings.setId(getInt(getColumnIndex(Cols.ID)));

        return settings;
    }
}

File: app/src/main/java/com/learning/ruslan/databases/SupportCursorWrapper.kt
package com.learning.ruslan.databases

//import com.learning.ruslan.databases.RusDbSchema.SupportTable
//import android.database.CursorWrapper
//import android.database.Cursor
//import com.learning.ruslan.*
//import com.learning.ruslan.settings.Settings
//
//class SupportCursorWrapper(cursor: Cursor?) : CursorWrapper(cursor) {
//    val settings: Settings
//        get() {
//            val settings = Settings()
//            settings[getInt(getColumnIndex(SupportTable.Cols.CHECKED)) == 1, getInt(getColumnIndex(
//                SupportTable.Cols.PAUSE)), getString(getColumnIndex(SupportTable.Cols.THEME)), getInt(
//                getColumnIndex(
//                    SupportTable.Cols.QUESTIONS))] =
//                getString(getColumnIndex(SupportTable.Cols.LANGUAGE))
//            settings.id = getInt(getColumnIndex(SupportTable.Cols.ID))
//            return settings
//        }
//}